test/e2e/util/version.go

```diff
@@ -61,6 +62,13 @@ func AtLeast(t *testing.T, version semver.Version) {
 	}
 }
 
+func CPOAtLeast(t *testing.T, version semver.Version, hc *hyperv1.HostedCluster) {
+	cpoVersion := semver.MustParse(hc.Status.Version.Desired.Version)
```

Comment:
I feel like we should panic here if `hc.Status.Version == nil` or `hc.Status.Version.Desired.Version == ""`

That means that `CPOAtLeast()` was called from a context where `WaitForImageRollout()` was not called first.

Reply:
This i ran as part of the t.after() function which runs only after the test and after the t.before() is called which calls the WaitForImageRoolout(). 

I will add nil checks just to make sure we don't panic and return some nice logs instead.